mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -2681,14 +2681,12 @@ SocketAsyncResult ValidateEndIAsyncResult (IAsyncResult ares, string methodName,
 
 		void QueueIOSelectorJob (SemaphoreSlim sem, IntPtr handle, IOSelectorJob job)
 		{
-			sem.WaitAsync ().ContinueWith (t => {
-				if (CleanedUp) {
-					job.MarkDisposed ();
-					return;
-				}
-
-				IOSelector.Add (handle, job);
-			});
+			sem.Wait();
```

Comment:
This is absolutely not going to work. The previous behaviour was purely async: the QueueIOSelectorJob would return before the `ContinueWith` callback would be called, and this `sem` would act as a queue. But now it's going to block until the `sem` is released. That effectively kills the asynchronousness of this method.